### PR TITLE
Nikola jokic/sync test check

### LINKED ISSUE #315: Write tests in yunikorn-core to make sure DAO snapshot is serialized the same way on REST endpoint and on the stream
This will ensure that future changes on the API are not unnoticed

### LINKED ISSUE #324: Separate every functional change in yunikorn-core fork to decrease chance of merge conflicts
When we sync fork in the future, we want to avoid having merge conflicts as much as we can. Since the intention is not to modify the scheduler behavior, the easiest way to decrease chance of merge conflicts is to move functionalities that we implemented into separate files.

Some changes will have to still exist inside the files forked from yunikorn-core. For example, fields added to the struct will still have to exist in files that will be synchronized with the upstream.

However, we can extract functionalities like snapshots and unlocked creation of dao objects by moving them to a separate file.

### PR DESCRIPTION
**Pull request by nikola-jokic**
Extract functions to avoid merge conflicts.

Fixes https://github.com/G-Research/unicorn-history-server/issues/315
Fixes https://github.com/G-Research/unicorn-history-server/issues/324

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -19,10 +19,13 @@
 
 module github.com/G-Research/yunikorn-core
 
-go 1.21
+go 1.22
+
+toolchain go1.23.2
```
Comment (richscott) [inline go.mod]: Do we want to require 1.23 for working in this repo? (I'm fine with doing that, just wondering if this is intentional.)

File: go.mod [hunk 2]
```diff
@@ -33,6 +36,7 @@ require (
 	github.com/prometheus/client_model v0.5.0
 	github.com/prometheus/common v0.45.0
 	github.com/sasha-s/go-deadlock v0.3.5
+	github.com/stretchr/testify v1.8.1
```
Comment (richscott) [inline go.mod]: Any interest in using `v1.9.0` (it's been out for many months)?

**Review by richscott — COMMENTED** [review]
I had just 2 questions/comments, but it otherwise looks great.

File: go.mod (hunk 1 above)
Comment (nikola-jokic) [reply go.mod]: Updated :relaxed:

**Review by richscott — APPROVED** [review]

**Pull request merged**